src/main/java/pl/fc/invoicing/services/impl/InvoiceServiceImpl.java

```diff
@@ -0,0 +1,79 @@
+package pl.fc.invoicing.services.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import pl.fc.invoicing.dto.InvoiceDto;
+import pl.fc.invoicing.dto.InvoiceListDto;
+import pl.fc.invoicing.model.Invoice;
+import pl.fc.invoicing.repositories.InvoiceRepository;
+import pl.fc.invoicing.services.InvoiceService;
+
+@Service
+@RequiredArgsConstructor
+public class InvoiceServiceImpl implements InvoiceService {
+
+    private final InvoiceRepository invoiceRepository;
+
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public InvoiceDto save(InvoiceDto invoiceDto) {
+        Invoice invoiceModel = modelMapper.map(invoiceDto, Invoice.class);
+        invoiceModel.updateRelations();
+        Invoice savedInvoice = invoiceRepository.save(invoiceModel);
+        return modelMapper.map(savedInvoice, InvoiceDto.class);
+    }
+
+    @Override
+    public Optional<InvoiceDto> getById(UUID id) {
+        Optional<Invoice> invoice = invoiceRepository.findById(id);
+        if (invoice.isEmpty()) {
+            return Optional.empty();
```

Comment:
you shouldn't return empty object. Just throw exception and handle it via ExceptionHandler

Reply:
Corrected.